src/anm/analytics.js

```diff
@@ -0,0 +1,76 @@
+var engine = require('engine');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = supportSendBeacon ? 2000 : 1000,
```

Comment:
>  The browser queues the request, and sends it “at the earliest available opportunity, but MAY prioritize the transmission of data lower compared to other network traffic.” (as per the W3C spec). sendBeacon returns `true` if the data was successfully submitted to the server, or `false` otherwise.
So you may use return value if it failed.